### DISCUSSION TITLE
feature: mappings to scroll commands

### DISCUSSION THREAD

**Discussion by Etavioxy** [vscode-neovim/vscode-neovim#1836] [Q&A]
### Did you check the docs?

- [X] I have read all the vscode-neovim docs
- [X] I understand the fundamental limitations of the plugin and believe this request to be reasonable

### Is your feature request related to a problem? Please describe.

https://github.com/vscode-neovim/vscode-neovim/blob/4a1b5fa18f569e9a495a6a82c25bed549e3b43ef/vim/vscode-scrolling.vim#L32-L46

i want to bind them to some keys, but the extension bind them directly. i can't find out how to bind them.

### Describe the solution you'd like

expose these commands as functions, or directly as original keys

### Describe alternatives you've considered

.

**Comment by Etavioxy** [comment]
I'm using this script now

```
no <expr> , char2nr(getcharsearch().char)==13 ? '<Cmd>lua require("vscode-neovim").call("editorScroll", {args={to="up",by="line",value=3}})<CR>' : ','
 no <expr> . char2nr(getcharsearch().char)==13 ? '<Cmd>lua require("vscode-neovim").call("editorScroll", {args={to="down",by="line",value=3}})<CR>' : ';'
```

**Comment by Guh-Feng** [comment]
Hello there, I wanted to rebind these keys too and took a look at some of the source files. I figured out that you have to delete the normal mode bindings in VSCode's Keyboard Shortcuts panel and then add them back in the kebindings.json file as shown below. The key was to change the "command" argument to "vscode-neovim.send" and add the "args" portion too.
```
{
        "key": "ctrl+e",
        "command": "vscode-neovim.send",
        "when": "editorTextFocus && neovim.ctrlKeysNormal.e && neovim.init && neovim.mode != 'insert' && editorLangId not in 'neovim.editorLangIdExclusions'",
        "args": "<C-e>"
    },
    {
        "key": "ctrl+d",
        "command": "vscode-neovim.send",
        "when": "editorTextFocus && neovim.ctrlKeysNormal.d && neovim.init && neovim.mode != 'insert' && editorLangId not in 'neovim.editorLangIdExclusions'",
        "args": "<C-d>"
    },
    {
        "key": "ctrl+f",
        "command": "vscode-neovim.send",
        "when": "editorTextFocus && neovim.ctrlKeysNormal.f && neovim.init && neovim.mode != 'insert' && editorLangId not in 'neovim.editorLangIdExclusions'",
        "args": "<C-f>"
    }
```
